docs/Measuring_Resonances.md

```diff
@@ -65,29 +65,29 @@ the system that may damage the electronics.
 
 ## Software installation
 
-Note that resonance measurements and shaper auto-calibration require additional
-software dependencies not installed by default. First, you will have to run on
-your Raspberry Pi the following command:
+Because setting up the software for auto-calibration takes significant
+time and resources, it is not installed by default. Run the following
+script to setup `numpy` and `matplotlib` to get started:
 ```
-~/klippy-env/bin/pip install -v numpy
+~/klipper/scripts/setup-shaper-calibrate.sh
 ```
-to install `numpy` package. Note that, depending on the performance of the
+Note that, depending on the performance of the
 CPU, it may take *a lot* of time, up to 10-20 minutes. Be patient and wait
 for the completion of the installation. On some occasions, if the board has
-too little RAM, the installation may fail and you will need to enable swap.
+too little RAM, the installation may fail, and you will need to
+[enable swap](https://nebl.io/neblio-university/enabling-increasing-raspberry-pi-swap/).
 
-Next, run the following commands to install the additional dependencies:
-```
-sudo apt update
-sudo apt install python-numpy python-matplotlib
-```
+## Hardware Configuration
+You can connect your ADXL345 to any spi bus supported by Klipper. Regular printer
```

Comment:
Unfortunately, "to any spi bus supported by Klipper" is not really correct. This is because ADXL345 is itself non-compliant to the SPI protocal, and therefore it cannot share a bus with any other device - otherwise there is a risk it will interfere with the communications with that device. On Raspberry Pi this is almost never an issue - there is almost never anything else connected to the SPI bus. However, on the printer MCU there is often something else - SD Card reader, SPI bus is shared with the TMC drivers, SPI goes to screen (where is also an SD Card reader), etc. Connecting ADXL345 to such SPI bus will lead to a confusing and surprising behavior. For example, connecting to the SPI bus shared with SD Card typically leads to boot failures and inability to install a firmware update (from the SD Card) for the boards that use SD Card method. Then, another big consideration is 5V tolerance. Most Chinese boards only accept 5V _as a power source_. They are not 5V tolerant on other pins, and connecting them to many printer MCU boards will simply fry them. ADXL345 from Adafruit and SparkFun have logic-level shifter built-in, so they are fine, but depending on the MCU, it could probably get the other way around and damage the MCU in turn (if the MCU SPI pins are not 5V tolerant). Overall, this is a complicated topic, and I see connecting the ADXL345 to a printer MCU only suitable for the expert users who can figure out those nuances. Otherwise, it's really playing a russian roulette - either it works and fine, or you fry some of your electronics.

So, I cannot recommend adding this and the MCU sections to the Klipper documentation as it is. These things need to be mentioned explicitly. Note that I was planning to write on this topic to the Klipper documentation, but didn't get to it yet (and I wouldn't be able to commit to some very near time). I'm more than happy if somebody else writes that, but I think those considerations and gotchas above need to be explicitly written to help users avoid damaging their hardware.

Reply:
I had no idea that the adxl345 had weird spi. :( I think the smart option at this point is to then have separate sections for RPi, secondary MCU, and primary MCU. Each would have it's own wiring directions, notes about power, and config snippets. RPi would be the recommended way, with secondary mcu recommended for if you don't want to install the linux mcu or you have something using your pi's gpio pins, and primary mcu for advanced users.

Reply:
Yep, unfortunately, it is like that. To be fair, primary and secondary MCU are almost equally tough, maybe with just less stuff connected to the secondary MCU. But it is less common scenario to have that secondary MCU in the first place. So we could have 2 different sections for primary/secondary MCUs, but it could also be a single section, because all the considerations and concerns are equally valid in both cases.